src/main/java/org/tikv/common/log/SlowLogImpl.java

```diff
@@ -68,45 +62,53 @@ public synchronized SlowLogSpan start(String name) {
     return slowLogSpan;
   }
 
+  @Override
+  public long getTraceId() {
+    return traceId;
+  }
+
+  @Override
+  public long getThresholdMS() {
+    return slowThresholdMS;
+  }
+
   @Override
   public void setError(Throwable err) {
     this.error = err;
   }
 
   @Override
   public void log() {
+    if (error != null || timeExceeded()) {
+      logger.warn("SlowLog:" + getSlowLogJson());
+    }
+  }
+
+  boolean timeExceeded() {
     long currentNS = System.nanoTime();
-    long currentMS = startMS + (currentNS - startNS) / 1_000_000;
-    if (error != null || (slowThresholdMS >= 0 && currentMS - startMS > slowThresholdMS)) {
-      logger.warn("SlowLog:" + getSlowLogString(currentMS));
+    long durationMS = (currentNS - startNS) / 1_000_000;
+    if (slowThresholdMS >= 0 && durationMS > slowThresholdMS) {
+      logger.warn("A slow request spent " + durationMS + "ms");
+      return true;
     }
+    return false;
   }
 
-  private String getSlowLogString(long currentMS) {
-    SimpleDateFormat dateFormat = getSimpleDateFormat();
+  JsonObject getSlowLogJson() {
     JsonObject jsonObject = new JsonObject();
 
-    jsonObject.addProperty("start", dateFormat.format(startMS));
```

Comment:
why remove the start and end time? how to know when a slow request is issued and when it's finished processing?

Reply:
All the information is moved to SlowLogSpan, see the [Proposal](https://pingcap.feishu.cn/docs/doccnyjWL25a7a0M8x8MGJ2wEWc)